packages/portal/src/app/homepage/homepage.component.ts

```diff
@@ -25,10 +27,49 @@ export class HomepageComponent implements OnInit {
   private listingService = inject(ListingService);
   private favoriteService = inject(FavoriteService);
   private userService = inject(UserService);
+  private realtimeService = inject(RealtimeService);
+
+  constructor(private snackBar: MatSnackBar) { }
+
+  notify(message: string) {
+    this.snackBar.open(message, "Close", {
+      duration: 10000,
+      horizontalPosition: 'end',
+      verticalPosition: 'top'
+    });
+  }
 
   async ngOnInit() {
     this.user = await this.userService.currentUser();
     this.featuredListings = await this.listingService.getFeaturedListings();
+
+    // Get notification when someone favourtied a listing
+    this.realtimeService.getNotifiedForFavourite((listingTitle) => {
+      const notifyMessage = this.user?.name?.length > 0
+        ? `Hurry up! Another user has favorited the listing "${listingTitle}".`
+        : `A user has favorited the listing "${listingTitle}".`;
+      this.notify(notifyMessage);
+    });
+
+    this.realtimeService.getNotifiedForCheckout(async (listing: Listing, from: string, to: string) => {
+      // Get notification when someone booked any listing
+      let notifyMessage = this.user?.name?.length > 0
+        ? `Hurry up! Another user has booked the listing "${listing.title}" but didn't complete the payment.`
+        : `A user has booked the listing "${listing}". but didn't complete the payment`;
+      this.notify(notifyMessage);
+
+      // Get notification when someone booked a listing which is favourited by current user
+      const favourites: Array<Listing> = await this.favoriteService.getFavoritesByUser(this.user) ?? [];
+
+      for (const favour of favourites) {
+        if (favour.id === listing.id) {
```

Comment:
this is just a styling suggestion but I feel the name is more representatitve. 'for (const fave of favorites) '